### PR TITLE
Fix #1815 Single Capture class created to handle all channels

### LINKED ISSUE #1806: wrong plotting w.r.t channel selection in Oscilloscope
**Actual Behaviour**

CH1 and CH3 are getting plotted in OscilloscopeActivity when CH2 and CH3 are selected.

CH1 and MIC input is getting plotted when MIC and CH2 is selected.

**Expected Behaviour**

CH1 and CH3 should be plotted when they are selected. 

**Steps to reproduce it**

Check if condition and corresponding captureTask.execute
https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/activity/OscilloscopeActivity.java#L309

Similar bug in this
https://github.com/fossasia/pslab-android/blob/development/app/src/main/java/io/pslab/activity/OscilloscopeActivity.java#L321

**Would you like to work on the issue?**

Yes

### LINKED ISSUE #1815: App crashes when user plays a saved gyroscope log
**Actual Behaviour**
App crashes when user plays a saved gyroscope log

**Expected Behaviour**
App should not crash

**Steps to reproduce it**
Record gyroscope data, and play it from data logger activity

**LogCat for the issue**
2019-06-21 22:19:31.751 17761-17761/io.pslab E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.pslab, PID: 17761
    java.lang.NullPointerException: Attempt to invoke virtual method 'int java.util.ArrayList.size()' on a null object reference
        at io.pslab.fragment.GyroscopeDataFragment.playData(GyroscopeDataFragment.java:286)
        at io.pslab.models.PSLabSensor.onOptionsItemSelected(PSLabSensor.java:339)
        at android.app.Activity.onMenuItemSelected(Activity.java:3469)
        at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:436)
        at android.support.v7.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:196)
        at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at android.support.v7.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:64)
        at android.support.v7.widget.Toolbar$1.onMenuItemClick(Toolbar.java:204)
        at android.support.v7.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:781)
        at android.support.v7.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:840)
        at android.support.v7.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
        at android.support.v7.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
        at android.support.v7.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:981)
        at android.support.v7.widget.ActionMenuView.invokeItem(ActionMenuView.java:625)
        at android.support.v7.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
        at android.view.View.performClick(View.java:6297)
        at android.view.View$PerformClick.run(View.java:24797)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6626)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)

**Would you like to work on the issue?**
Yes

### LINKED ISSUE #1816: Convert Multiple CaptureTasks in Oscilloscope to a Single Capture Task
**Actual Behaviour**
Currently there are multiple CaptureTasks which has lot of redundant code.

**Expected Behaviour**
Make a single CaptureTask to handle work of all the other capture tasks

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1816 Fixes #1806

**Changes**: Single Capture class created to handle all channels

**Screenshot/s for the changes**:

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[capture_task_refactor.zip](https://github.com/fossasia/pslab-android/files/3316811/capture_task_refactor.zip)

@CloudyPadmal  @mariobehling  have a look at this. The code looks a lot cleaner now. Will do similar for fft and Audiobuffer tasks in subsequent PRs.

### CODE REVIEW

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java [hunk 1]
```diff
@@ -1074,143 +1085,62 @@ public boolean onTouchEvent(MotionEvent event) {
...
     public class CaptureTask extends AsyncTask<String, Void, Void> {
-        ArrayList<Entry> entries;
-        String analogInput;
+        ArrayList<ArrayList<Entry>> entries = new ArrayList<>();
+        Integer noOfChannels;
+        String[] paramsChannels;
 
         @Override
-        protected Void doInBackground(String... params) {
+        protected Void doInBackground(String... strings) {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Rename `strings` to `channels`. It makes more sense.

File: app/src/main/java/io/pslab/activity/OscilloscopeActivity.java [hunk 2]
```diff
@@ -1074,143 +1085,62 @@ public boolean onTouchEvent(MotionEvent event) {
...
+            paramsChannels = strings;
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Need to copy the variable?

Comment (neel1998) [reply app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Yes... Because I am using input parameters ```strings``` in onPostExecute() method

Comment (CloudyPadmal) [reply app/src/main/java/io/pslab/activity/OscilloscopeActivity.java]: Ahh I see..

**Comment by neel1998** [comment]
@CloudyPadmal  please check. And kindly merge if looks fine because I want to solve further issues in oscilloscope. Thanks

**Pull request merged**